### PR TITLE
enable Dashboard iframe loading with flag

### PR DESCRIPTION
**Pull request by leiyanggz**
The value of the response header "X-Frame-Options" is "deny" by default in http requests because of security concerns. 
However in some use cases people want to load the dashboard in their internal web applications via iframe.  

# Changes

Add an additional boolean flag to disable the header "X-Frame-Options".

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/main/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Review by AlanGreene — CHANGES_REQUESTED** [review]
Thanks for the PR @leiyanggz. I've left a suggestion to make this a little more flexible while still allowing users to disable it. Let me know if you have any questions.

File: cmd/dashboard/main.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ var (
 	logFormat          = flag.String("log-format", "json", "Format for log output (json or console)")
 	streamLogs         = flag.Bool("stream-logs", false, "Enable log streaming instead of polling")
 	externalLogs       = flag.String("external-logs", "", "External logs provider url")
+	enableXframe       = flag.Bool("enable-xframe", false, "Whether or not a browser should be allowed to render a page in a <frame>, <iframe>, <embed> or <object>")
```
[the remark this answers is not in the document]
Comment (leiyanggz) [reply cmd/dashboard/main.go:50]: Got it. Will do.

[the remark this answers is not in the document]
Comment (leiyanggz) [reply cmd/dashboard/main.go]: Please take a look at the new pull request https://github.com/tektoncd/dashboard/pull/2104

**Comment by leiyanggz** [comment]
Close the pull request since new pull request submitted https://github.com/tektoncd/dashboard/pull/2104

**Pull request closed without merging**